tutorial/source/vae.ipynb

```diff
@@ -85,13 +85,47 @@
     "\n",
     "So much for preliminaries. Let's see how we implement a VAE in Pyro.\n",
     "The dataset we're going to model is MNIST, a collection of images of handwritten digits.\n",
-    "Since this is a popular benchmark dataset, we can make use of PyTorch's convenient data loader functionalities to reduce the amount of boilerplate code we need to write:\n",
+    "Since this is a popular benchmark dataset, we can make use of PyTorch's convenient data loader functionalities to reduce the amount of boilerplate code we need to write:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
     "\n",
-    "```python\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
     "import torch\n",
     "import torchvision.datasets as dset\n",
+    "import torch.nn as nn\n",
     "import torchvision.transforms as transforms\n",
     "\n",
+    "import pyro\n",
+    "import pyro.distributions as dist\n",
+    "from pyro.infer import SVI, Trace_ELBO\n",
+    "from pyro.optim import Adam"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pyro.enable_validation(False)\n",
```

Comment:
Should we set `pyro.enable_validation(smoke_test)` to be safe?

Reply:
Ok. This would be a nice use case for #988 
```py
pyro.sample("obs", dist.Bernoulli(prediction), obs=dist.Bernoulli(ground_truth))
```
or even for parallel enumeration (if it doesn't add too much overhead):
```py
truth = pyro.sample("truth", dist.Bernoulli(ground_truth),
                    infer={"enumerate": "parallel"})
pyro.sample("obs", dist.Bernoulli(prediction), obs=truth)
```
But we can address this later...